pyproject.toml

```
@@ -1,10 +1,225 @@
+]
+
+all-non-platform = [
+    "twisted[test,tls,conch,serial,http2,contextvars]",
+]
+
+macos-platform = [
+    "twisted[all-non-platform]",
+    "pyobjc-core",
+    "pyobjc-framework-CFNetwork",
+    "pyobjc-framework-Cocoa",
+]
+
+windows-platform = [
+    "twisted[all-non-platform]",
+    "pywin32 != 226",
+]
+
+osx-platform = [
+    "twisted[macos-platform]",
+]
+
+gtk-platform = [
+    "twisted[all-non-platform]",
+    "pygobject",
+]
+
+mypy = [
+    "twisted[dev,all-non-platform,conch-nacl]",
+    "mypy==0.930",
+    "mypy-zope==0.3.4",
+    "types-setuptools",
+    "types-pyOpenSSL",
+]
+
+[project.scripts]
+cftp = "twisted.conch.scripts.cftp:run"
+ckeygen = "twisted.conch.scripts.ckeygen:run"
+conch = "twisted.conch.scripts.conch:run"
+mailmail = "twisted.mail.scripts.mailmail:run"
+pyhtmlizer = "twisted.scripts.htmlizer:run"
+tkconch = "twisted.conch.scripts.tkconch:run"
+trial = "twisted.scripts.trial:run"
+twist = "twisted.application.twist._twist:Twist.main"
+twistd = "twisted.scripts.twistd:run"
+
+[project.urls]
+Changelog = "https://github.com/twisted/twisted/blob/HEAD/NEWS.rst"
+Documentation = "https://docs.twistedmatrix.com/"
+Homepage = "https://twistedmatrix.com/"
+Issues = "https://twistedmatrix.com/trac/report"
+Source = "https://github.com/twisted/twisted"
+Twitter = "https://twitter.com/twistedmatrix"
+
+[tool.hatch.version]
+source = "code"
+path = "src/twisted/__init__.py"
+search-paths = ["src"]
+
+[tool.hatch.metadata.hooks.fancy-pypi-readme]
+content-type = "text/x-rst"
+fragments = [
+    { path = "README.rst" },
+]
+
+[[tool.hatch.metadata.hooks.fancy-pypi-readme.substitutions]]
+# Munge links of the form `NEWS <NEWS.rst>`_ to point at the appropriate
+# location on GitHub so that they function when the long description is
+# displayed on PyPI.
+pattern = '`([^`]+)\s+<(?!https?://)([^>]+)>`_'
+replacement = '`\1 <https://github.com/twisted/twisted/blob/trunk/\2>`_'
+ignore-case = true
+
+[tool.hatch.build.targets.wheel]
+packages = [
+    "src/twisted",
+]
+exclude = [
+    "*.c",
+    "*.h",
+    "*.pxi",
+    "*.pyx",
+    "build.bat",
+]
+
+[tool.hatch.build.targets.sdist]
+# This dictates what goes into a sdist tarball.
+# As a release tarball, we want to include:
+#  - All of the library code.
+#  - All the useful documentation (but not the super historic stuff)
+#  - All of the test files, so that the tests can be run on an installed copy.
+# We do not want to include:
+#  - Release management files (e.g. topfiles)
+#  - Things only useful when running from a source checkout
+include = [
+    # Documentation
+    "/docs",
+
+    # Library code
+    "/src",
+    "/CONTRIBUTING.md",
+    "/INSTALL.rst",
+    "/NEWS.rst",
+    "/README.rst",
+    "/code_of_conduct.md",
+
+    # Test-running utilities for downstream packagers
+    "/.coveragerc",
+    "/mypy.ini",
+    "/tox.ini",
+]
+exclude = [
+    # Do not include the news fragments
+    "/src/twisted/newsfragments",
```

There are a bunch more newsfragment directories that need to be listed here.

---

Sorry, where?

---

I don't know if we should care... for a release, these folders should be empty.

The fragment folders are only populated on the dev branches.

```sh
$ find src/ -name newsfragments
src/twisted/conch/newsfragments
src/twisted/trial/newsfragments
src/twisted/mail/newsfragments
src/twisted/newsfragments
src/twisted/persisted/newsfragments
src/twisted/names/newsfragments
src/twisted/words/newsfragments
src/twisted/scripts/newsfragments
src/twisted/runner/newsfragments
src/twisted/application/newsfragments
src/twisted/web/newsfragments
```

---

@adiroiban To the extent that there is a supported workflow for fixing stuff upstream and then upgrading to a release when it's available, it is to point your application's `requirements.txt` at a dev branch, or at least a snapshot on trunk, so this should still be fixed.  Here's a suggestion for easy committing:

```suggestion
    "src/twisted/conch/newsfragments",
    "src/twisted/trial/newsfragments",
    "src/twisted/mail/newsfragments",
    "src/twisted/newsfragments",
    "src/twisted/persisted/newsfragments",
    "src/twisted/names/newsfragments",
    "src/twisted/words/newsfragments",
    "src/twisted/scripts/newsfragments",
    "src/twisted/runner/newsfragments",
    "src/twisted/application/newsfragments",
    "src/twisted/web/newsfragments",
```